cli/cmd/upgrade.go

```diff
@@ -251,6 +251,13 @@ func upgrade(ctx context.Context, k *k8s.KubernetesAPI, flags []flag.Flag, stage
 			return bytes.Buffer{}, err
 		}
 	}
+	// if there is still no linkerd-config-overrides secret, Linkerd has been installed using
+	// Helm installation, where linkerd-config-overrides secret doesn't exist
+	// User must be notified to upgrade Linkerd using Helm
+	if values == nil {
+		err := errors.New("Linkerd was installed with Helm Installation, upgrade Linkerd with Helm")
```

Comment:
I'm not sure we want to say with complete certainty that Linkerd was installed with Helm at this point; maybe the installation is just in a very odd state.

It may be worth changing the messaging on this to something like:

```suggestion
		err := errors.New("failed to fetch current install configuration; if Linkerd was installed with Helm, users need to upgrade with Helm.")
```

The comment should also be updated with similar messaging.

Reply:
:sweat_smile: I am not good with writing such errors, I am replacing it, with what you have suggested